pkg/envoy/grpc.go

```diff
@@ -50,14 +50,30 @@ func (s *xdsServer) startXDSGRPCServer(listener net.Listener, config map[string]
 
 	reflection.Register(grpcServer)
 
+	ctx, cancel := context.WithCancel(context.Background())
 	go func() {
+		if s.restorerPromise != nil {
+			restorer, err := s.restorerPromise.Await(ctx)
+			if err == nil && restorer != nil {
+				log.Infof("Envoy: Waiting for endpoint restoration before serving xDS resources...")
+				err = restorer.WaitForEndpointRestore(ctx)
```

Comment:
Quick question, to confirm - is this err value that is intended to be checked below against `context.Canceled`?